stader-cli/validator/deposit.go

```diff
@@ -52,22 +54,82 @@ func nodeDeposit(c *cli.Context) error {
 		}
 	}
 
-	canNodeDepositResponse, err := staderClient.CanNodeDeposit(baseAmount, big.NewInt(int64(numValidators)), true)
+	// Get node status
+	status, err := staderClient.NodeStatus()
 	if err != nil {
 		return err
 	}
+
+	fmt.Printf(
+		"The node %s%s%s has a balance of %.6f SD.\n\n",
+		log.ColorBlue,
+		status.AccountAddress,
+		log.ColorReset,
+		math.RoundDown(eth.WeiToEth(status.AccountBalances.Sd), eth.Decimal))
+
+	// Get node SD status
+	sdStatusResp, err := staderClient.GetSDStatus()
+	if err != nil {
+		return err
+	}
+
+	sdStatus := sdStatusResp.SDStatus
+	amountToCollateral := new(big.Int).Sub(sdStatus.SdCollateralRequireAmount, sdStatus.SdCollateralCurrentAmount)
+
+	utilityAmount := big.NewInt(0)
+
+	if sdStatus.NotEnoughSdCollateral {
+		fmt.Printf(
+			"The node %s%s%s had not enough SD in collateral please deposit SD.\n\n",
+			log.ColorBlue,
+			status.AccountAddress,
+			log.ColorReset,
+		)
+
+		ops := []string{"Deposit from node wallet", "Utility SD"}
+		i, _ := cliutils.Select("Choose option", ops)
+
+		switch i {
+		case 0:
+			if status.AccountBalances.Sd.Cmp(amountToCollateral) < 0 {
+				cliutils.PrintError("You do not had enough SD in your wallet. Please deposit and try again")
+				return nil
+			}
+
+			err = node.DepositSdWithAmount(staderClient, amountToCollateral, autoConfirm, 0)
+			if err != nil {
+				return err
+			}
+
+			return nodeDeposit(c)
```

Comment:
i think we can just let the flow continue after this rather then re-calling the function.